### PR TITLE
Fix UDC platform App crash `Can't create handler inside thread` during PromptForCommissionOKPermission

### PR DESCRIPTION
**Pull request by nicelyjust**
When send UDC message to the platform App，met an error

```
D  CDC msg sent
D  ------PROMPT USER: Test TV casting app is requesting permission to cast to this TV, approve? [0x0000_FFF1,0x0000_8001,F67F0EDFD6DF2741,M}]
D  Received prompt for OK permission vendor id:65521 productId:32769. Commissionee: Test TV casting app
E  Java exception in PromptForCommissionOKPermission
W  java.lang.RuntimeException: Can't create handler inside thread Thread[CHIP AppServer IO Thread,10,main] that has not called Looper.prepare()
W      at android.os.Handler.<init>(Handler.java:227)
W      at android.os.Handler.<init>(Handler.java:129)
W      at android.app.Dialog.<init>(Dialog.java:147)
W      at android.app.Dialog.<init>(Dialog.java:199)
W      at androidx.appcompat.app.AppCompatDialog.<init>(AppCompatDialog.java:57)
W      at androidx.appcompat.app.AlertDialog.<init>(AlertDialog.java:98)
W      at androidx.appcompat.app.AlertDialog$Builder.create(AlertDialog.java:983)
W      at com.matter.tv.server.MatterCommissioningPrompter.promptForCommissionOkPermission(MatterCommissioningPrompter.java:64)
E  PromptForCommissionOKPermission error: ../../examples/tv-app/android/java/MyUserPrompter-JNI.cpp:99: CHIP Error 0x00000003: Incorrect state
D  ------Via Shell Enter: controller ux ok|cancel
```

and refactor the MatterCommissioningPrompter class,It should not obtained activty instance.

### CODE REVIEW

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public class ContentAppAgentService extends Service {
   public static final int FAILED_UNKNOWN = 0x01;
   public static final int FAILED_TIMEOUT = 0x94;
 
-  private static final int COMMAND_TIMEOUT = 8; // seconds
+  private static final int COMMAND_TIMEOUT = 4; // seconds
```
Comment (chrisdecenzo) [inline examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java:39]: was this change intentional? Can you provide a comment explaining why 4 seconds and not 8?

**Review by chrisdecenzo — APPROVED** [review]

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java (hunk 1 above)
Comment (andy31415) [reply examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java:39]: if we are touching this, I recommend adding units of measurements for everything that is a unitless integer types. This should be `COMMAND_TIMEOUT_SEC` and below should be `ATTRIBUTE_TIMEOUT_SEC` or you should convert these to a Duration type.

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java [hunk 2]
```diff
@@ -144,4 +155,43 @@ private void createNotificationChannel() {
       notificationManager.createNotificationChannel(channel);
     }
   }
+
+  static class VH extends Handler{
```
Comment (andy31415) [inline examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java:159]: please extend the name to something more expressive than `VH`

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java [hunk 3]
```diff
@@ -22,109 +39,103 @@ public class MatterCommissioningPrompter extends UserPrompterResolver implements
...
+                    + " productId:"
+                    + productId
+                    + ". Commissionee: "
+                    + commissioneeName);
+
+    Message message = Message.obtain();
+    message.arg1 = vendorId;
+    message.arg2 = productId;
+    message.obj = commissioneeName;
+    message.what = 101;
```
Comment (andy31415) [inline examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java:65]: could we have a constant for these magic numbers?

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java [hunk 4]
```diff
@@ -22,109 +39,103 @@ public class MatterCommissioningPrompter extends UserPrompterResolver implements
...
+                    + vendorId
+                    + " productId:"
+                    + productId
+                    + ". Commissionee: "
+                    + commissioneeName);
+
+    Message message = Message.obtain();
+    message.arg1 = vendorId;
+    message.arg2 = productId;
+    message.obj = commissioneeName;
```
Comment (andy31415) [inline examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java:64]: could we encapsulate all data together if we send an object anyway? Why not have a class that contains all of vendorID, productId and name. That way you do not need to remember arg1 being vendor and arg2 being product and naming becomes more safe.

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java [hunk 5]
```diff
@@ -144,4 +155,43 @@ private void createNotificationChannel() {
...
+  static class VH extends Handler{
+
+    private final WeakReference<MatterCommissioningPrompter> mPrompterWeakReference;
+    public VH(MatterCommissioningPrompter commissioningPrompter){
+      super(Looper.getMainLooper());
+      mPrompterWeakReference = new WeakReference<>(commissioningPrompter);
+    }
+    @Override
+    public void handleMessage(@NonNull Message msg) {
+      MatterCommissioningPrompter commissioningPrompter = mPrompterWeakReference.get();
```
Comment (andy31415) [inline examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java:168]: arg1 and arg2 seem unused here. Is there a plan to use them? If not, maybe don't need to set them.

File: examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java (hunk 1 above)
Comment (nicelyjust) [reply examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java:39]: @chrisdecenzo  the client  send UDC message to the platform App in main thread，So I lowered the value to avoid ANR

Comment (nicelyjust) [reply examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java:39]: @andy31415 yes，we did do [that](https://github.com/project-chip/connectedhomeip/blob/11ea4316f3e5a60f8d34fcad8deeb154803498a8/examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java#L151)

**Review by chrisdecenzo — APPROVED** [review]
